src/youtube.coffee

```diff
@@ -24,11 +27,21 @@ module.exports = (robot) ->
         key: process.env.HUBOT_YOUTUBE_API_KEY
       })
       .get() (err, res, body) ->
-        videos = JSON.parse(body)
+        robot.logger.debug body
+        if err
+          robot.logger.error err
+          return msg.send err
+        try
+          videos = JSON.parse(body)
```

Comment:
Most of the times JSON is going to fail is when it wasn't a 200 error. It'd be good to check `res.statusCode is 200` before trying to parse, and return an error otherwise.